delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/files/Files.scala

```diff
@@ -138,6 +186,38 @@ final class Files(
     } yield res
   }.named("updateFile", moduleType)
 
+  /**
+    * Update a new file linking it from an existing file in a storage
+    *
+    * @param id         the file identifier to expand as the iri of the file
+    * @param storageId  the optional storage identifier to expand as the id of the storage. When None, the default storage is used
+    * @param projectRef the project where the file will belong
+    * @param rev        the current revision of the file
+    * @param filename   the optional filename to use
+    * @param mediaType  the optional media type to use
+    * @param path       the path where the file is located inside the storage
+    */
+  def updateLink(
+      id: IdSegment,
+      storageId: Option[IdSegment],
+      projectRef: ProjectRef,
+      filename: Option[String],
+      mediaType: Option[ContentType],
+      path: Uri.Path,
+      rev: Long
+  )(implicit caller: Caller): IO[FileRejection, FileResource] = {
+    for {
+      project               <- projects.fetchActiveProject(projectRef)
+      iri                   <- expandIri(id, project)
+      _                     <- test(UpdateFile(iri, projectRef, testStorageRef, testStorageType, testAttributes, rev, caller.subject))
+      (storageRef, storage) <- fetchActiveStorage(storageId, project)
+      resolvedFilename      <- IO.fromOption(filename.orElse(path.lastSegment), InvalidFileLink(iri))
+      description           <- FileDescription(resolvedFilename, mediaType)
+      attributes            <- storage.moveFile(path, description).leftMap(MoveRejection(iri, storage.id, _))
+      res                   <- eval(UpdateFile(iri, projectRef, storageRef, storage.tpe, attributes, rev, caller.subject), project)
```

Comment:
In order to avoid STM code duplication and in order to make it more similar to V.1.4.x there is no different event.
At the end of the day it is the same: creation or update of a file. The file is not coming from the client payload but coming from a storage. But the end result is the same: the FileAttributes being presented

Reply:
I don't see any advantage from doing that. It just brings code duplication on the STM side and makes more different than 1.4.x.

At the end of the day it is the same: creation or update of a file. The file is not coming from the client payload but coming from a storage. But the end result is the same: the FileAttributes being presented.

Reply:
Creating a different event may be too much but the main idea was to be able to know how the file was created for debugging purposes for example

Reply:
The current event contains enough information for you to know how the file was created:
- storageRev
- storageType
- fileAttributes (if the Digest in the file attributes is empty, it is a file linking)

Reply:
The last one is not very explicit and may be true today but may change tomorrow as it may be interesting for a user to be able to verify that a downloaded file is correct

Reply:
Or for the DLM part that is coming in v1.6 where we will move file around, it may be good to have digests for them too.

Reply:
I've added a field to `FileAttributes` to know the origin of the attributes.